core/math/linalg/general.odin

```diff
@@ -212,7 +212,13 @@ matrix_mul_differ :: proc "contextless" (a: $A/matrix[$I, $J]$E, b: $B/matrix[J,
 
 
 @(require_results)
-matrix_mul_vector :: proc "contextless" (a: $A/matrix[$I, $J]$E, b: $B/[J]E) -> (c: B)
+matrix_mul_vector :: proc "contextless" (a: $A/matrix[$I, I]$E, b: $B/[I]E) -> (c: B)
+	where !IS_ARRAY(E), IS_NUMERIC(E) #no_bounds_check {
+	return a * b
+}
+
+@(require_results)
+matrix_mul_vector_differ :: proc "contextless" (a: $A/matrix[$I, $J]$E, b: $B/[J]E) -> (c: matrix[I, 1]E)
```

Comment:
Have you even tested this code to see if it even works?

Reply:
Yes but see I missed copying something over. Sorry about that.